conf/config.go

```diff
@@ -122,40 +122,49 @@ func LoadConfig(mainConfigFile string, fallbackConfigFile string) (*MenderConfig
 		return nil, loadErr
 	}
 
+	log.Debugf("Loaded %d configuration file(s)", filesLoadedCount)
+
 	if filesLoadedCount == 0 {
 		log.Info("No configuration files present. Using defaults")
 		return config, nil
 	}
 
-	if config.Servers == nil {
-		if config.ServerURL == "" {
+	log.Debugf("Loaded configuration = %#v", config)
+
+	return config, nil
+}
+
+// VerifyServerConfig verifies the Servers fields in the configuration
+func (c *MenderConfig) VerifyServerConfig() error {
+	if c.Servers == nil {
+		if c.ServerURL == "" {
 			log.Warn("No server URL(s) specified in mender configuration.")
 		}
-		config.Servers = make([]client.MenderServer, 1)
-		config.Servers[0].ServerURL = config.ServerURL
-	} else if config.ServerURL != "" {
+		c.Servers = make([]client.MenderServer, 1)
+		c.Servers[0].ServerURL = c.ServerURL
+	} else if c.ServerURL != "" {
 		log.Error("In mender.conf: don't specify both Servers field " +
 			"AND the corresponding fields in base structure (i.e. " +
 			"ServerURL). The first server on the list on the" +
 			"list overwrites these fields.")
-		return nil, errors.New("Both Servers AND ServerURL given in " +
+		return errors.New("Both Servers AND ServerURL given in " +
 			"mender.conf")
 	}
-	for i := 0; i < len(config.Servers); i++ {
+	for i := 0; i < len(c.Servers); i++ {
 		// Trim possible '/' suffix, which is added back in URL path
-		if strings.HasSuffix(config.Servers[i].ServerURL, "/") {
-			config.Servers[i].ServerURL =
+		if strings.HasSuffix(c.Servers[i].ServerURL, "/") {
```

Comment:
You may also want to move this part as well to `loadConfig`.

Reply:
And this part I guess we could move it also, but I was thinking about an scenario like: the user load something (invalid), does some modifications in memory, then validates. Moving this to load, the possible modification will be invalid (like url's with trailing `/`)

Reply:
ahh, I only meant the part trimming the prefix (which only silently corrects the loaded config). But you may ignore this since we can't really do anything with similar correction above.